cumulus/client/cirrus-executor/src/bundle_processor.rs

```diff
@@ -547,13 +564,51 @@ where
 				))
 			})?;
 
-			// TODO: generate_proof(trace_mismatch_index, local_receipt, bad_signed_receipt_hash)
-			// and submit
+			let fraud_proof = self
+				.fraud_proof_generator
+				.generate_proof::<PBlock>(
+					trace_mismatch_index as usize,
+					&local_receipt,
+					bad_signed_receipt_hash,
+				)
+				.map_err(|err| {
+					sp_blockchain::Error::Application(Box::from(format!(
+						"Failed to generate fraud proof: {err}"
+					)))
+				})?;
+
+			self.submit_fraud_proof(fraud_proof);
 		}
 
 		Ok(())
 	}
 
+	fn submit_fraud_proof(&self, fraud_proof: FraudProof) {
+		let primary_chain_client = self.primary_chain_client.clone();
+		// TODO: No backpressure
+		self.spawner.spawn_blocking(
+			"cirrus-submit-fraud-proof",
+			None,
+			async move {
+				tracing::debug!(
+					target: LOG_TARGET,
+					"Submitting fraud proof in a background task..."
+				);
+				if let Err(error) = primary_chain_client.runtime_api().submit_fraud_proof_unsigned(
+					&BlockId::Hash(primary_chain_client.info().best_hash),
+					fraud_proof,
+				) {
+					tracing::debug!(
```

Comment:
Isn't this an error?

Reply:
This will be resolved with the formatting PR https://github.com/subspace/subspace/pull/652#discussion_r913524487 altogether later.